src/jobflow_remote/jobs/daemon.py

```diff
@@ -405,65 +600,156 @@ def start(
+                # old configuration.
+                old_config = self.parse_config_file()
+                new_config = {
+                    "num_procs_transfer": num_procs_transfer,
+                    "num_procs_complete": num_procs_complete,
+                    "single": single,
+                    "log_level": log_level,
+                    "connect_interactive": connect_interactive,
+                }
+                diffs_config = []
+                for config_name, config_value in new_config.items():
+                    if config_value != old_config.get(config_name):
+                        diffs_config.append(config_name)
+                if diffs_config:
+                    logger.warning(
+                        f"Daemon is {DaemonStatus.STOPPED.value}, but the options {', '.join(diffs_config)} "
+                        "differ from the values used to activate supervisor. The daemon will start with the initial "
+                        "configurations. To change the configuration shut down the daemon and start it again."
+                    )
+                error = self.start_processes()
+            elif status == DaemonStatus.STOPPING:
+                error = "Daemon process are stopping. Cannot start."
+            else:
+                error = f"Daemon status {status} could not be handled"
+
+            if error is not None:
+                if raise_on_error:
+                    raise DaemonError(error)
+                logger.error(error)
+                return False
+            lock.update_on_release = {
+                "$set": {"running_runner": self._get_runner_info()}
+            }
+            return True
 
     def stop(self, wait: bool = False, raise_on_error: bool = False) -> bool:
-        status = self.check_status()
-        if status in (
-            DaemonStatus.STOPPED,
-            DaemonStatus.STOPPING,
-            DaemonStatus.SHUT_DOWN,
-        ):
-            return True
+        """
+        Stop the daemon.
+
+        Parameters
+        ----------
+        wait
+            If True wait until the daemon has stopped.
+        raise_on_error
+            Raise an exception if the daemon cannot be stopped.
+
+        Returns
+        -------
+        bool
+            True if the daemon was stopped successfully, False otherwise.
+        """
+        with self.lock_runner_doc() as lock:
+            doc_error = self._check_running_runner(
+                lock.locked_document, raise_on_error=raise_on_error
+            )
+            if doc_error:
+                logger.error(doc_error)
+                return False
+            status = self.check_status()
+            if status in (
+                DaemonStatus.STOPPED,
+                DaemonStatus.STOPPING,
+                DaemonStatus.SHUT_DOWN,
+            ):
+                lock.update_on_release = {"$set": {"running_runner": None}}
+                return True
```

Comment:
When it's "STOPPING", are we sure that all processes will indeed be stopped upon releasing the lock ? Could there be any problem here ?

Reply:
Indeed it cannot be guaranteed. However, following what we discussed about the fact that also when running the `jf runner stop/shutdown` it cannot be guaranteed that the runner will be actually stopped immediately, I think it should be fine to leave it this way also for the `STOPPING` state.

Reply:
Yes, ok